web_ui/packages/ui/src/virtualize-horizontal-grid/horizontal-layout.ts

```diff
@@ -0,0 +1,82 @@
+// Copyright (C) 2022-2025 Intel Corporation
+// LIMITED EDGE SOFTWARE DISTRIBUTION LICENSE
+
+import { InvalidationContext } from '@react-stately/virtualizer';
+import { Key, Layout, LayoutInfo, Rect, Size } from 'react-aria-components';
+
+const DEFAULT_SIZE = 100;
+
+export interface HorizontalLayoutOptions {
+    gap?: number;
+    size?: number;
+}
+
+export class HorizontalLayout extends Layout {
+    protected gap: number;
+    protected size: number;
+
+    constructor(options: HorizontalLayoutOptions = {}) {
+        super();
+        this.gap = options.gap ?? 8;
+        this.size = options.size ?? DEFAULT_SIZE;
+    }
+
+    // Determine which items are visible within the given rectangle.
+    getVisibleLayoutInfos(rect: Rect): LayoutInfo[] {
+        if (!this.virtualizer) {
+            return [];
+        }
+
+        const sizeWithGap = this.size + this.gap;
+        const keys = Array.from(this.virtualizer.collection.getKeys());
+        const startIndex = Math.max(0, Math.floor(rect.x / sizeWithGap));
+        const endIndex = Math.min(keys.length - 1, Math.ceil(rect.maxX / sizeWithGap));
+        const layoutInfos = [];
```

Comment:
```suggestion
        const layoutInfos: LayoutInfo[] = [];
```
to avoid the cast  on line 50